solutions/usaco-809.mdx

```diff
@@ -2,14 +2,98 @@
 id: usaco-809
 source: USACO Bronze 2018 Febuary
 title: Taming the Herd
-author: Sathvik Chundru
+author: Sathvik Chundru, Aditya Gupta
 ---
 
 [Official Analysis (C++)](http://www.usaco.org/current/data/sol_taming_bronze_feb18.html)
 
 **Time Complexity:** $\mathcal{O}(N)$
 
 <LanguageSection>
+	
+<CPPSection>
+
+```cpp
+#include <iostream>
+#include <bits/stdc++.h>
+
+using namespace std;
+
+int main() {
+	freopen("taming.in", "r", stdin);
+	freopen("taming.out", "w", stdout);
+	int n;
+	cin >> n;
+	/*
+	 * A status of 2 means a breakout happened on day n.
+	 * A status of 1 means a breakout did not happen on day n.
+	 * A status of 0 means we are not sure what happened on day n.
+	 */
+	vector<int> status(n,0);
+	status[0] == 2;  // a breakout must occur on the first day
+	vector<int> log(n);  // log for each day
+	for (int i = 0; i < n; i++) {
+		cin >> log[i];
+	}
+	bool valid = true;
+	if (log[0] == 1) {
+		valid = false; // we found a discrepancy
+	} else {
+		status[0] = 2; // the problem says the first day had a breakout
+	}
+	
+	// we go through our log, updating our status array accordingly
+	for (int i = 1; i < n; i++) {
+		if (!valid) break;
+		int curr_n = log[i];  // the current day's log
```

Comment:
What would you like me to change it to?

Reply:
whatever you think better represents the variable is tracking than ```curr_n```

Reply:
Alright, I don't even think we need a variable actually lol

Reply:
i think it’s better if you do use a variable because it’ll be easier to see what’s going on. @SansPapyrus683 do you have any thoughts?

Reply:
I feel like its fairly obvious in the problem what the purpose of using the log[i] is, and when it's not, I provided comments. Don't you think? If not, then, please tell me what you want the variable name to be, since it seems relatively easy to understand for the reader.